File: DungeonDebt/Assets/Scripts/Data/RivalGuildState.cs
public class RivalGuildState
{
    public RivalGuildState(
        RivalGuild guild,
        string displayName,
        int morale,
        int debt,
        int payroll,
        string statusLabel,
        int payrollGrowthPerRound)
    {
        Guild = guild;
        DisplayName = displayName;
        Morale = morale;
        Debt = debt;
        Payroll = payroll;
        StatusLabel = statusLabel;
        PayrollGrowthPerRound = payrollGrowthPerRound;
    }

    public RivalGuild Guild { get; set; }
    public string DisplayName { get; set; }
    public int Morale { get; set; }
    public int Debt { get; set; }
    public int Payroll { get; set; }
    public string StatusLabel { get; set; }
    public int PayrollGrowthPerRound { get; set; }
}

File: web/src/data/RivalGuildState.js
// Ported from DungeonDebt/Assets/Scripts/Data/RivalGuildState.cs
// Mutable rival snapshot.
export class RivalGuildState {
  constructor(guild, displayName, morale, debt, payroll, statusLabel, payrollGrowthPerRound) {
    this.guild = guild;
    this.displayName = displayName;
    this.morale = morale;
    this.debt = debt;
    this.payroll = payroll;
    this.statusLabel = statusLabel;
    this.payrollGrowthPerRound = payrollGrowthPerRound;
  }
}
